### DISCUSSION TITLE
Error: System.NotSupportedException: DeserializeNoConstructor

### DISCUSSION THREAD

**Discussion by AngxiaoLu123** [elringus/bootsharp#96] [Q&A] [answered]
Hi, this is a great tool, I want to ask does it support passing the file changed event from js to C#? I tried but got this: 

Error: System.NotSupportedException: DeserializeNoConstructor, JsonConstructorAttribute, Microsoft.AspNetCore.Components.Forms.InputFileChangeEventArgs Path: $ | LineNumber: 0 | BytePositionInLine: 1.

**Comment by elringus** [comment, marked as answer]
Only JSON-serializable types can be passed directly (by value), but it's also possible to interop with instances via wrappers:

https://github.com/Elringus/DotNetJS/blob/a1720e7819e3b06b78c1100a042e7e02cc1dd6c2/JavaScript/test/interop.js#L85-L114

**Reply by AngxiaoLu123** [reply]
Sorry, how can I wrap InputFileChangeEventArgs to instances? I want to unzip user selected file and upload files in the webassembly without Blazor, I tried .Net 7 but JSExport attribute does not support this. Do you think it is possible? Thanks a lot!

**Reply by elringus** [reply]
Only .NET 6 is supported for the time being: #20
